### PR TITLE
feat: transaction store and toasts

### LINKED ISSUE #1679: Vega transaction toasts
## Story

**As a** user
**I want** vega transactions to be shown as toasts
**So that** I can continue to use the app whilst waiting for a tx to complete

## Acceptance Criteria

- [x] I am prompted via dialog to complete the transaction in wallet
- [ ] I am immediately shown a rejected transaction (when rejected by wallet immediately)
- [x] If the transaction was accepted by the wallet, I can still use the application and am not blocked by an overlay
- [x] If the transaction is accepted I am shown a toast with the tx hash, indicating its pending
- [ ] I can execute multiple transactions one after the other
- [ ] When a transaction result is received, notify state by toast

## Tasks

- [ ] Create transaction store
- [ ] Migrate to using toasts
  - [ ] https://github.com/vegaprotocol/frontend-monorepo/issues/2271
  - [ ] https://github.com/vegaprotocol/frontend-monorepo/issues/2270
  - [ ] Withdrawal creation issue TBD

## Additional information

- We may want to create a new hook/store for this. Say a useVegaTransactionV2 to make development and migration to the new pattern easier and more incremental

### PR DESCRIPTION
**Pull request by asiaznik**
# Related issues 🔗

Closes #1679 

# Demo 📺

https://user-images.githubusercontent.com/1980305/206734301-b4b953e1-2254-43b2-a1ca-5d731eb84ce6.mov

### CODE REVIEW

**Review by MadalinaRaicu — APPROVED** [review]

File: libs/wallet/src/use-vega-transaction-store.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      const transaction: VegaStoredTxState = {
+        id: transactions.length,
+        createdAt: now,
+        updatedAt: now,
+        body,
+        error: null,
+        txHash: null,
+        signature: null,
+        status: VegaTxStatus.Requested,
+        dialogOpen: true,
```
Comment (mattrussell36) [inline libs/wallet/src/use-vega-transaction-store.tsx:67]: question (non blocking): why do we need to track dialogOpen state?

File: libs/withdraws/src/lib/create-withdrawal-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    (state) => state.isOpen
+  );
+  return (
+    <Dialog
+      title={t('Withdraw')}
+      open={isOpen && !connectWalletDialogIsOpen}
+      onChange={(isOpen) => (isOpen ? open() : close())}
+      size="small"
+    >
+      <Web3Container connectEagerly>
```
Comment (mattrussell36) [inline libs/withdraws/src/lib/create-withdrawal-dialog.tsx:23]: chore (blocking): No need to use Web3Container here anymore

File: libs/withdraws/src/lib/pending-withdrawals-table.tsx [hunk 3]
```diff
@@ -13,138 +13,107 @@ import type {
...
+        >) => {
+          return value ? getDateTimeFormat().format(new Date(value)) : '';
+        }}
+      />
+      <AgGridColumn
+        headerName=""
+        field="status"
+        flex={2}
+        cellRendererParams={{
+          complete: async (withdrawal: WithdrawalFieldsFragment) => {
```
Comment (mattrussell36) [inline libs/withdraws/src/lib/pending-withdrawals-table.tsx:110]: chore (non blocking): unnecessary async

File: libs/withdraws/src/lib/withdrawal-feedback.tsx [hunk 4]
```diff
@@ -15,6 +15,25 @@ import { useWeb3React } from '@web3-react/core';
 import { formatDistanceToNow } from 'date-fns';
 import type { WithdrawalFieldsFragment } from './__generated__/Withdrawal';
 
+/*
```
Comment (mattrussell36) [inline libs/withdraws/src/lib/withdrawal-feedback.tsx:18]: chore (non blocking): can we clear this out if its not needed

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Big chunk of work here! My main blocking chain is the use of Web3Containers everywhere, which shouldn't be required anymore as we now have the Web3ReactProvider wrapping the entire app

**Comment by mattrussell36** [comment]
Just noticed this z-index issue with the order book header

Screenshot 2022-12-14 at 21 47 01

File: libs/wallet/src/use-vega-transaction-store.tsx (hunk 1 above)
Comment (bglownia) [reply libs/wallet/src/use-vega-transaction-store.tsx:67]: dialogOpen indicates it dialog connected with stored transaction should be visible to the user. It controls with toast notification is open. Will be set to false if user will dismiss it. 
It was not renamed to keep a consistency.

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**